scripts/profiler/events.py

```diff
@@ -42,109 +41,47 @@ def deserialize(json):
 
 
 class TrackedEvent():
+    TRACKED_EVENT_FIELDNAMES = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+
     def __init__(self, submit, start_time, end_time):
         self.submit = submit
         self.proc_start_time = start_time
         self.proc_end_time = end_time
 
+    def serialize(self):
+        si = StringIO()
```

Comment:
I am wondering if it would be possible to keep some of the objects instead of creating and destroying them with each event.
`si` looks like something that could be class-level variable
maybe `wr` can be one too? (`si.truncate(0)` could help)

maybe something similar can be done with deserialize

Reply:
just a hint - not need if not possible to be done or gives no performance benefit

Reply:
btw for reader you can do
```
si.seek(0);si.truncate();si.write('test3');si.seek(0)
```
`list(rd)` will return updated stuff

Reply:
Those methods are elements of TrackedEvent class. And in an object of such class serialize method is executed once or twice maximum. Deserialize method is a static method that is used to create an object of TrackedEvent class. The only way I see to avoid creating multiple times objects that you mentioned is to create new class that would be responsible for serialization and deserialization. Those methods of such class would take respectively TrackedEvent object or string as an argument. I just wonder if it's worth and if it won't be too complicated. What do you think about it?

Reply:
discussed offline